inspirehep/modules/workflows/tasks/matching.py

```diff
@@ -257,20 +274,36 @@ def raise_if_match_wf_in_error_or_initial(obj, eng):
         None
 
     """
-    def _filter(base_record, match_result):
-        return get_value(
-            match_result, '_source._workflow.status'
-        ) in ('ERROR', 'INITIAL')
-
-    matched_ids = pending_in_holding_pen(obj, _filter)
+    matched_ids = set_wf_not_completed_ids_to_wf(obj)
     if bool(matched_ids):
-        obj.extra_data['error_workflows_matched'] = matched_ids
         raise WorkflowsError(
-            'Cannot continue processing. Found workflows in ERROR or INITIAL '
-            'state: {}'.format(matched_ids)
+            'Cannot continue processing workflow {wf_id}.'
+            'Found not-completed workflows in holdingpen '
+            ': {blocking_ids}'.format(wf_id=obj.id,
+                                      blocking_ids=matched_ids)
         )
 
 
+def remove_blocked_by_me(obj, matched_ids):
```

Comment:
Don't you think this should be named something like `get_blocking_workflows_ids`, specifying in the docstring that cycles are avoided?
It does not remove anything from anywhere :reminder_ribbon: